### PR TITLE
Adds dueling pistols.

### PR DESCRIPTION
**Pull request by AnturK**
Why is our gun code so backwards. Anyway, adds dueling pistols for the times when you need something more classy than toolbox to the head for solving disputes. Could use some unique sprites.

Quick guide how this works:
- Both parties grab the gun and move 5 steps away.
- They pick one of three projectile modes.
- Countdown starts.
- When fired gun creates chaff that blocks opposite projectile type.
- Losers lose their heads.

### CODE REVIEW

**Comment by WJohn** [comment]
Maybe also add this in the holodeck as well? Could be cool to resolve fights that way without anyone dying, but that's up to you.

**Comment by AnturK** [comment]
I'd have to map holodeck area and that's above my effort quota for this week.

**Comment by OverlyGeneric** [comment]
I wouldn't say a bullet would decapitate someone but I would say the brain trauma caused by it would render the person unclonable.

Here are the old-fashioned rules to dueling despite this being _glorified RPS_.

- The person who felt offended or dishonored had to "challenge" his opponent to a duel and this was usually done by throwing his glove down in front of the opponent or by hitting him in the face with a glove. 

- Each person had to then find a person to be his "second" and the job of the seconds was to choose a place to duel and to decide whether or not the weapons were equal.

- The person who declared the duel got to choose when the duel would be finished. Common endings were if one person was injured, grievously injured or hurt, and killed. 

- Duels with pistols could be ended after the first shot, even if no one was hit. Most pistol duels didn't go longer than three shots.

Techno-flintlocks for sprites? 🗿

**Comment by ATH1909** [comment]
https://www.youtube.com/watch?v=m7iHmuco_zo

**Comment by Repukan** [comment]
If a person doesn't want to die they can just not duel.

Kinda stupid to make something thematically lethal non lethal

**Comment by SgtxRootbeer** [comment]
@Repukan
Because sometimes roleplayers might want something else at stake for the duel.
Eg:
Racist human vs lizard. If the Liz wins, the human has to say on comms that lizards are the Master race, if the human wins, then the lizard has to build a gas chamber for himself.
Assistant vs rd: if the assistant wins, then the Rd will remove the DNA lock on his phazon and give it to the assistant, if the Rd wins, the assistant feeds the slimes in xenobio.

**Comment by Repukan** [comment]
Sounds stupid

**Comment by PKPenguin321** [comment]
Duels are typically the final lethal level of escalation, sgtx. You use them when you want to kill somebody

**Comment by FantasticFwoosh** [comment]
Even if its non-lethal, nothing says that the captain can't decree you to be executed or punished anyway if either participant loses this duel for their innocence. Least its more resolute than trial by combat which is messy.

**Comment by ATH1909** [comment]
Add a function where if you're on help intent and fire your dueling pistol, your character intentionally make your shot miss (or just not count as a hit if it connects with your target? I dunno).

File: code/modules/projectiles/guns/energy/dueling.dm [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+		return BULLET_ACT_BLOCK
+	
+	var/obj/item/bodypart/B = L.get_bodypart(BODY_ZONE_HEAD)
+	B.dismember()
+	qdel(B)
+
+//Storage case.
+/obj/item/storage/lockbox/dueling
+	name = "dueling pistol case"
+	desc = "Let's solve this like gentlemen."
```
Comment (MrDoomBringer) [inline code/modules/projectiles/guns/energy/dueling.dm]:
```suggestion
	desc = "Let's solve this like gentlespacemen."
```

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

**Comment by MrDoomBringer** [comment]
image
new sprites by some loser

**Comment by OverlyGeneric** [comment]
> new sprites by some loser
> color changes based on firing mode

very good. feels like it belongs in a Steampunk setting but fits just as well in a futuristic setting.

**Comment by ATH1909** [comment]
> > new sprites by some loser
> > color changes based on firing mode
> 
> very good. feels like it belongs in a Steampunk setting but fits just as well in a futuristic setting.

It reminds me of the laser muskets from the Fallout series.

**Comment by ExcessiveUseOfCobblestone** [comment]
I want this but don't want to review this :^)

**Comment by ExcessiveUseOfCobblestone** [comment]
nvm it looks fine, are you going to change the sprite

**Pull request merged**